libraries/AP_Airspeed/AP_Airspeed.cpp

```diff
@@ -399,16 +408,90 @@ void AP_Airspeed::read(void)
     }
 #endif
 
-    // setup primary
-    if (healthy(primary_sensor.get())) {
-        primary = primary_sensor.get();
-        return;
-    }
-    for (uint8_t i=0; i<AIRSPEED_MAX_SENSORS; i++) {
-        if (healthy(i)) {
-            primary = i;
-            break;
+    switch(blend_conf) {
+    case BLEND_NONE:
+        // setup primary
+        if (healthy(primary_sensor.get())) {
+            primary = primary_sensor.get();
+        } else {
+            for (uint8_t i=0; i<AIRSPEED_MAX_SENSORS; i++) {
+                if (healthy(i)) {
+                    primary = i;
+                    break;
+                }
+            }
+        }
+        break;
+    case BLEND_FULL:
+        // TODO: check have_inertial_nav()?
+
+        // TODO: check _ahrs != NULL
+
+        // blend the airspeed from several sensors and EKF
+        Vector3f velocity, wind;
+        if (_ahrs->get_velocity_NED(velocity) == false) {
+            // TODO: correct calculation
+            return;
         }
+        wind = _ahrs->wind_estimate();
+        velocity -= wind;
+        float airspeed_ekf = velocity.length();
+
+        float airspeed_avg = 0.0;
+        float ekf_sensor_airspeed_diff[AIRSPEED_MAX_SENSORS];
+        float avg_sensor_airspeed_diff[AIRSPEED_MAX_SENSORS];
+        for (uint8_t i = 0; i < AIRSPEED_MAX_SENSORS; i++) {
+            airspeed_avg += state[i].airspeed;
+        }
+        airspeed_avg /= AIRSPEED_MAX_SENSORS;
+
+        // calculating inverse square of error that will be our trust in sensor
```

Comment:
We need to simulate the two sensor case in SITL due to the possibility of problems caused by circular data dependancy with the EKF.

Reply:
Agree, I am looking into that, though the support for several sensors initially was made before this PR.